sec/review.tex

```diff
@@ -321,6 +327,9 @@ \subsubsection{Hierarchical clustering}
 choosing an appropriate number of clusters; a decision that is usually made post
 hoc according to another metric, such as the silhouette coefficient, evaluated
 at each level of the tree.
+% TODO I wonder if a figure  with a tree would be nice here (on an actual data
+% set). And likewise we could have one of those classic k-means scatter plots in
+% the previous section?
```

Comment:
Agreed. Those k-means plots also demonstrate the Voronoi tessellation nicely if I can add in the halfplanes.